Refresh.GameServer/Services/RequestStatisticTrackingService.cs

```diff
@@ -4,28 +4,30 @@
 using Bunkum.Core.Responses;
 using Bunkum.Core.Services;
 using NotEnoughLogs;
-using Refresh.GameServer.Database;
 using Refresh.GameServer.Endpoints;
 
 namespace Refresh.GameServer.Services;
 
 public class RequestStatisticTrackingService : Service
 {
+    private static int _gameRequestsToSubmit;
+    private static int _apiRequestsToSubmit;
+    
+    public static (int game, int api) SubmitAndClearRequests()
+    {
+        return (game: Interlocked.Exchange(ref _gameRequestsToSubmit, 0),
+                api: Interlocked.Exchange(ref _apiRequestsToSubmit, 0));
+    }
+    
     internal RequestStatisticTrackingService(Logger logger) : base(logger)
     {}
 
     public override Response? OnRequestHandled(ListenerContext context, MethodInfo method, Lazy<IDatabaseContext> database)
     {
-        GameDatabaseContext gameDatabase = (GameDatabaseContext)database.Value;
-
         if (context.Uri.AbsolutePath.StartsWith(GameEndpointAttribute.BaseRoute))
-        {
-            gameDatabase.IncrementGameRequests();
-        }
+            Interlocked.Exchange(ref _gameRequestsToSubmit, _gameRequestsToSubmit + 1);
         else
-        {
-            gameDatabase.IncrementApiRequests();
-        }
+            Interlocked.Exchange(ref _apiRequestsToSubmit, _apiRequestsToSubmit + 1);
```

Comment:
These should be using `Interlocked.Increment(ref value)`
this will be compiled down to the operations
- read i32 into register (this part is atomic)
- increment register
- exchange value with the incremented register (this part is atomic)

since the whole operation is not done atomically, theres the chance for race conditions

while interlocked.increment will be compiled down to the operation
- increment value at address (atomic)

Reply:
oh. yeah, i did not read the second comment properly.